app/jobs/alert_email/base.rb

```diff
@@ -1,19 +1,20 @@
 class AlertEmail::Base < ApplicationJob
+  MAXIMUM_RESULTS_PER_RUN = 500
+
   def perform
     return if DisableExpensiveJobs.enabled?
 
-    subscriptions.find_each do |subscription|
-      next if subscription.alert_run_today?
+    # The intent here is that if we don't have keyword or location searches, then this operation can all be done in memory
+    # really fast (1 week's worth of vacancies is around 2000, so not worth leaving on disk for each of 100k daily subscriptions
+    default_scope = Vacancy.includes(:organisations).live.order(publish_on: :desc).search_by_filter(from_date: from_date, to_date: Date.current)
+
+    already_run_ids = AlertRun.for_today.map(&:subscription_id)
```

Comment:
I think we can avoid a step here an directly query the ids from DB:
```suggestion
    already_run_ids = AlertRun.for_today.pluck(:subscription_id)
```